gwcli/gateway.py

```diff
@@ -865,6 +865,19 @@ def ui_command_create(self, gateway_name, ip_addresses, nosync=False,
 
         api = APIRequest('{}/sysinfo/hostname'.format(new_gw_endpoint))
         api.get()
+
+        if api.response.status_code != 200:
```

Comment:
Nit: any chance this `hostname` API get could be moved to before the attempted GW creation? From a quick glance it looks like it's just pulling the hostname so if it could do that first you could avoid the need to attempt to delete the gateway.

Reply:
Good idea! Let me find out if we can do it. Will update the patch soon. Thanks @dillaman

Reply:
Done